### PR TITLE
SB-850: Nuget `delete` command implemented

### PR DESCRIPTION
**Pull request by sbespalov**

### CODE REVIEW

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetPackageController.java [hunk 1]
```diff
@@ -97,6 +98,38 @@
     @Inject
     private NugetSearchPackageSource packageSource;
     
+    @RequestMapping(path = { "{storageId}/{repositoryId}/{packageId}/{versionString}" }, method = RequestMethod.DELETE)
+    @PreAuthorize("hasAuthority('ARTIFACTS_DEPLOY')")
+    public ResponseEntity deletePackageNew(@RequestHeader(name = "X-NuGet-ApiKey", required = false) String apiKey,
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetPackageController.java]: Couldn't this method simply be called `delete`...?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetPackageController.java [hunk 2]
```diff
@@ -97,6 +98,38 @@
...
+                                           @ApiParam(value = "The storageId", required = true) @PathVariable(name = "storageId") String storageId,
+                                           @ApiParam(value = "The repositoryId", required = true) @PathVariable(name = "repositoryId") String repositoryId,
+                                           @PathParam("packageId") String packageId,
+                                           @PathParam("versionString") String versionString)
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetPackageController.java]: Could you please change this to:
```
@PathParam("version") String version)
```

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/nuget/NugetPackageControllerTest.java [hunk 3]
```diff
@@ -1,35 +1,51 @@
...
+/**
+ * @author Sergey Bespalov
+ *
+ */
 @IntegrationTest
 @RunWith(SpringJUnit4ClassRunner.class)
 public class NugetPackageControllerTest extends RestAssuredBaseTest
 {
 
+    private static final String API_KEY = "eyJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJTdHJvbmdib3giLCJqdGkiOiJ0SExSbWU4eFJOSnJjNXVXdTVkZDhRIiwic3ViIjoiYWRtaW4iLCJzZWN1cml0eS10b2tlbi1rZXkiOiJhZG1pbi1zZWNyZXQifQ.xRWxXt5yob5qcHjsvV1YsyfY3C-XFt9oKPABY0tYx88";
```
Comment (carlspring) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/nuget/NugetPackageControllerTest.java]: I'm not sure, if you've documented this, but -- how do you, (as a user), get this API key? Do you get it from the REST API (we don't have a UI yet, so I assume "yes"), or what...?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetPackageController.java (hunk 1 above)
Comment (sbespalov) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetPackageController.java]: yes, will be renamed

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetPackageController.java (hunk 2 above)
Comment (sbespalov) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetPackageController.java]: ok

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/nuget/NugetPackageControllerTest.java (hunk 3 above)
Comment (sbespalov) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/nuget/NugetPackageControllerTest.java]: it described here:
https://github.com/sbespalov/strongbox-examples/tree/sbespalov_SB-850/hello-strongbox-nuget-mono#set-api-key-to-use-with-your-repository

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/AbstractArtifactManagementService.java [hunk 4]
```diff
@@ -391,4 +392,29 @@ public InputStream resolve(String storageId,
         return null;
     }
 
-}
+    @Override
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/AbstractArtifactManagementService.java]: It was a great idea to extract this here, thanks! :)

**Pull request merged**